go/ql/lib/semmle/go/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -344,6 +358,9 @@ class DataFlowCall extends Expr {
     not exists(this.getEnclosingFunction()) and result.asFileScope() = this.getFile()
   }
 
+  /** Gets the location of this call. */
+  Location getLocation() { result = super.getLocation() }
```

Comment:
I wondered why this was needed. On commenting it out I found that the super impl returns `DbLocation`, whereas `Location = LocationImpl` and `LocationImpl` is a newtype made up of `DbLocation` and synthetic locations. It then wasn't immediately obvious to me why the above line compiles. Then I found `class DbLocationImpl extends LocationImpl instanceof DbLocation`. So if I've understood correctly, this predicate is needed because the super impl doesn't satisfy the needed signature.

Reply:
Correct.